### ISSUE TITLE
Suggestion for user files

### ISSUE THREAD

**Issue by rickteuthof** [leegbestand/coconut#17]
We could have the user traversal and pass files be stored in a subdirectory under the root directory of the user project, alongside the main. Here we can have a pregenerated cmakelists containing all the pass and traveral files the target compiler depends on. We can also add a user compiler flag that allows the user to specify their project directory, in which the pass and traversal files will be pregenerated. Of course it should only be necessary to run this once.

In addition to this, i think we should drop support for specifying a specific source and include directory for the generated files, as this isn't really necessary.

the structure will look something like this:

```
user_project 
| - user_src/
| | - pass_something.c
| | - trav_something.c
| | - CMakeLists.txt (pre-generated)
| - main.c
```

**Comment by leegbestand** [comment]
Probably easiest is to make no choice about the directory of the user. Just provide the option to generate user files and let the user specify an output directory(the source and include directory option can work like that and remove the original idea). This way the user can decide where the files should be generated. Also, it might be nice to give the name(maybe a list) of pass/traversal as an additional option so that only that one gets generated.

**Comment by rickteuthof** [comment]
Yes that is exactly what i was aiming for. I do like the addition of specifying a single file, which would solve the problem for when new traversals are added to the dsl.
